artifacts-wg/charter/charter.md

```diff
@@ -0,0 +1,49 @@
+# Purpose
+In the distributed world of cloud-native computing different artifacts and packages are used to transport configuration and code for the many services and capabilities that comprise and support workloads and applications. As an example, today ArtifactHub advertises that it can crawl about 20 such artifact types [1]. These "cloud-native" bundles add to the previous proliferation of package and artifact types for software dependencies such as Maven and npm and for system packages like deb and rpm.
+
+This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully learned and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development by users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
+
+To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects offering packaging, distribution and deployment mechanisms for bundles of configuration and code.
+
+**The WG will:**  
+* **Develop common patterns**  
+*  **Advocate for innovative projects**  
+*  **Gather end user feedback**
+**for packaging, discovery, distribution, and deployment mechanisms for** **cloud-native software artifacts.**
+
+# Alignment with TAG App Delivery and CNCF
+TAG App Delivery's charter [2] calls on the TAG to develop best practices, foster collaboration between projects, improve interoperability and propose new initiatives and projects related to the following domains which this working group would target.
+
+* Application bundling and deployment
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on official specifications in the Open Containers Initiative (OCI) itself.
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/artifacts
```

Comment:
I'm expecting this repo to be archived in the not too distant future:
https://github.com/opencontainers/artifacts/issues/61
https://github.com/opencontainers/artifacts/issues/63

We may want to link to image-spec instead.

Reply:
Having this group demonstrates theres been a lot of value in OCI Artifacts, so it's not clear why we would archive the repo.

Reply:
I think that discussion is best had in the linked issues. The short version is that image-spec has added a lot of details on how to package content other than images, and it has made changes that aren't included in the artifact repo. Rather than trying to keep the two in sync, I think it makes sense to consolidate to one repo.

Reply:
I've removed the reference altogether in my re-imagining of the goals section.

Reply:
That,

1) all "artifacts" use OCI artifacts path and by extension use the dist-spec referrers facility

- OR -

2) all "artifacts" use https://github.com/opencontainers/artifacts (or derivative) and/or implement specs coming out of this WG

remains unresolved IMO.

I suspect this may just turn out to be a cost-benefit business decision in the end. Too costly to do 2) so just do 1) for example. But not sure we want to make this assumption so early in the game.

Reply:
Thanks for this discussion, it's helpful to understand the potential issues with existing artifacts and the evolution of image-spec. Thanks for participating in this group @sudo-bmitch.